modules/contacts/server/controllers/contacts.server.controller.js

```diff
@@ -45,6 +48,29 @@ exports.add = function(req, res) {
         });
       }
 
+      // Check if contact already exists
+      Contact.findOne({
+        $or: [
+          { users: [ req.body.friendUserId, req.user._id ] },
+          { users: [ req.user._id, req.body.friendUserId ] }
+        ]
+      }).exec(function(err, existingContact) {
+        if (err) return done(err);
+
+        if (existingContact) {
+          // Contact already exists!
+          return res.status(400).json({
+            message: errorHandler.getErrorMessageByKey('invalid-id')
```

Comment:
Actually, if users are already connected, it shouldn't yeld scary error at the frontend (error at the API is good), but rather just semi success informing something like "Hey great, you guys are already connected!"